_topics/en/op_codeseparator.md

```diff
@@ -0,0 +1,150 @@
+---
+title: OP_CODESEPARATOR
+
+## Optional.  An entry will be added to the topics index for each alias
+#aliases:
+#  - Foo
+
+## Required.  At least one category to which this topic belongs.  See
+## schema for options
+categories:
+  - Scripts and Addresses
+
+## Required.  Use Markdown formatting.  Only one paragraph.  No links allowed.
+## Should be less than 500 characters
+excerpt: >
+  **OP_CODESEPARATOR** is an opcode that changes what data is used when
+  a signature commits to a script.  The opcode has been available since
+  the original version of Bitcoin Script, but its use and behavior have
+  changed over time (with further changes having been proposed).
+
+## Optional.  Use Markdown formatting.  Multiple paragraphs.  Links allowed.
+extended_summary: |
+  In Bitcoin 0.1, each spend's scriptSig and outpoint's scriptPubKey
+  were evaluated together with a [virtual][virtual codesep]
+  `OP_CODESEPARATOR` placed between them.  For example, using a P2PK
+  script:
+
+      <signature> OP_CODESEPARATOR <public key> OP_CHECKSIG
+
+  In legacy Bitcoin Script, such as the above example,
+  `OP_CODESEPARATOR` removes itself and any preceding parts of a script
+  from the *scriptCode*---which is part of the data that is hashed to
+  create the message that is signed by a signature in Bitcoin.  The
+  above code separator allows the signature to only commit to the
+  `<pubkey> OP_CHECKSIG` part of the script.  Some operation of this
+  type is required because the signature can't commit to
+  itself---although Bitcoin 0.1 also directly removed any signatures
+  from the commitment using a function named `FindAndDelete()`.
+
+  There's [speculation][todd speculation] that the code separator was
+  also part of an earlier Bitcoin design that could have allowed coin
+  delegation.  For example, Alice owns a coin but adds Bob to the set of
+  people who can spend it---without Alice giving away her private key or
+  creating an onchain transaction.  Bob can then delegate further by
+  adding Carol to the set of allowed spenders.  No released version of
+  Bitcoin supported this capability but source code claimed to be from
+  a pre-release version of Bitcoin [may][pre todd1] have [supported][pre
+  todd2] this capability.
+
+  In [Bitcoin 0.3.7][] (July 2010), one of the fixes related to the [1 OP_RETURN][] bug was
+  deployed as a stealth [hard fork][lukejr says sipa confirms hf] that
+  changed how the scripts were evaluated, including discontinuing the
+  use of the virtual `OP_CODESEPARATOR`.  However, the opcode itself
+  wasn't changed and scripts have continued to be able to include it
+  directly.
+
+  ### Later changes and proposals
+
+  The [BIP143][] signature verification rules for the initial version of
```

Comment:
s/the initial version of segwit (v0)/segwit v0/ (initial version makes this sound to me like a pre-release version that was changed before being activated)

Reply:
updated